core/trino-spi/src/main/java/io/trino/spi/security/Identity.java

```diff
@@ -130,13 +130,17 @@ public boolean equals(Object o)
             return false;
         }
         Identity identity = (Identity) o;
-        return Objects.equals(user, identity.user);
+        return Objects.equals(user, identity.user) &&
+               Objects.equals(groups, identity.groups) &&
+               Objects.equals(principal, identity.principal) &&
+               Objects.equals(enabledRoles, identity.enabledRoles) &&
```

Comment:
From commit message

> The `StatementAnalyzer`, during the analysis stage, collects all table
references along with their corresponding access controls and
identities. At the end of the analysis, this is used to do the access
checks. They are collected in a `Map`, where the key is the pair of
access control and `Identity`, so that we deduplicate the necessary
checks. But for this deduplication to work properly, `Identity`
instances that result in different accesses should be considered
distinct.


is this backed by any tests? i.e. any tests that would involve StatementAnalyzer?

Reply:
Also, this looks orthogonal to "Always create new identity for view owner" commit.
correct me if am wrong, but maybe this could be separate PR?